plugins/modules/kube.py

```diff
@@ -0,0 +1,357 @@
+#!/usr/bin/python
```

Comment:
I don't see this as removed elsewhere are we supposed to have two versions of this now?

Reply:
Unfortunately, the answer is yes. We would have to maintain a duplicate of this since an Ansible collection structure places libraries under `plugins`, see [here](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections_structure.html#collection-directories-and-files), while an Ansible playbook structure 
places libraries under `library`, see [here](https://docs.ansible.com/ansible/2.8/user_guide/playbooks_best_practices.html#directory-layout).

The idea of this first iteration is to retain the structure as-is, so users could either use the repository as an Ansible collection, by simply pointing their requirements.txt directly at the Kubespray's repository, or download it and use it the same as before. And this seemed like the easiest way forward.

Reply:
Random guess but did you try to see if a symlink works here?

Reply:
It would _probably_ work, but before I start changing anything, do we want to go down this path? Symlinks will probably work fine on Linux distributions, and should work out-of-the-box when cloning (not tested), but I believe that symlinks in Windows require admin rights to access mklink functions plus enabling `core.symlinks` functionality when cloning, which might result in some confusion for users running Ansible from Windows. There might even be some additional edge cases that I'm not familiar with. Are we comfortable with this?

Reply:
Windows without WSL is not natively supported as a control node.
https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#control-node-requirements

Reply:
The issue with Windows and symlinks isn't related to Kubernetes nodes, but rather from which operating system the user is cloning the repository. 

I would assume some users are cloning the repository locally to a Windows OS and then, from Windows, deploying Kubernetes.

Reply:
That settles it :) I'll see if I can't get the collection working with a symlink instead of duplicating the library. Thanks for the feedback.

Reply:
Btw. how about using `#!/usr/bin/env bash` everywhere or at least in all new places introduced with this PR? This way operating systems like NixOS are supported, too. Using `env` is the best way when going for maximum portability. https://www.gnu.org/software/coreutils/manual/html_node/env-invocation.html

Reply:
@MrFreezeex, good suggestion! Worked like a charm! Let me know if there's any other concerns regarding this PR.